### PR TITLE
Update help text and handling of subscription ID

### PR DESCRIPTION
**Pull request by mboersma**
**Description**

I noticed we were missing help strings for a few arguments in `az capi create`. 

Also we provided the `--subscription-id` argument but didn't do the same for the other required authentication fields. I can't see why `AZURE_SUBSCRIPTION_ID` needed this special handling, so I removed it to be consistent.

**History Notes**

---

This checklist is used to make sure that common guidelines for an Azure CLI pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).
- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).
- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by mboersma** [comment]
Apologies, logically this should be two separate PRs, but in my defense it's all related to the help text and it's a small changeset.

**Review by DannyBrito — APPROVED** [review]
lgtm

File: src/capi/azext_capi/_help.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
     short-summary: Use ephemeral disks
   - name: --external-cloud-provider
     type: bool
+    short-summary: Use an external cloud provider component
```
Comment (CecileRobertMichon) [inline src/capi/azext_capi/_help.py:37]:
```suggestion
    short-summary: Use the external (AKA "out-of-tree") Azure cloud-provider 
```

a bit more precise

**Review by CecileRobertMichon — APPROVED** [review]
lgtm

**Pull request merged**